common/utils.js

```diff
@@ -1339,6 +1339,33 @@
             }
         }
 
+        function overrideDownloadClickBehavior(containerElem) {
+            var downloadElements = containerElem[0].querySelectorAll(".deriva-url-validate");
+
+            downloadElements.forEach(function (el) {
+                angular.element(el).on('click', function (e) {
+                    e.preventDefault();
+
+                    var spinnerHTML = ' <span class="glyphicon glyphicon-refresh glyphicon-refresh-animate"></span>';
+                    //show spinner
+                    e.target.innerHTML += spinnerHTML;
+
+                    var dcctx = ConfigUtils.getContextJSON();
+                    // make a HEAD request to check if the user can fetch the file
+                    dcctx.server.http.head(e.target.href).then(function (response) {
+                        // fetch the file for the user
+                        $window.open(e.target.href, '_blank');
```

Comment:
This will open the file in a new tab which is not the same as the default behavior. You could do a simple `$window.location = e.target.href`.

Although this means that we're not going to trigger the browser's default download behavior. I don't know if we can or whether this can cause issues.

Reply:
We might want to look into whether we can manually trigger the same event that the download does.

Reply:
The way this change behaves is much more similar to the browser's default behavior.

Reply:
much more as compared to which other solution? The browser's default behavior is not opening a new tab. I noticed this because after a few downloads my browser disallowed opening new tabs and asked for my permission.

Reply:
I meant to say that your suggested change is more similar to what the browser does by default (if not that same), so I'm going to update that.